### ISSUE TITLE
InvalidCastException in CSharpPreferIsEmptyOverCountFixer

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#3798]
@Jozkee I am seeing the following exception when attempting to FixAll `CA1836` violations on roslyn-analyzers repo. Is this something that will be addressed by this PR, or would you like me to file an issue?

```
System.InvalidCastException : Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.ArgumentSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.BinaryExpressionSyntax'.
   at Microsoft.NetCore.CSharp.Analyzers.Performance.CSharpPreferIsEmptyOverCountFixer.GetMemberAccessorFromBinary(SyntaxNode binaryExpression,Boolean useRightSide)
   at async Microsoft.NetCore.Analyzers.Performance.PreferIsEmptyOverCountFixer.<>c__DisplayClass3_0.<RegisterCodeFixesAsync>b__0(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionInternalAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.BatchFixAllProvider.GetChangedDocumentsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.BatchFixAllProvider.GetDocumentIdToChangedDocumentsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.BatchFixAllProvider.TryMergeFixesAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.BatchFixAllProvider.TryGetMergedFixAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.BatchFixAllProvider.GetFixAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.BatchFixAllProvider.GetFixAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.FixAllGetFixesService.GetFixAllCodeActionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.FixAllGetFixesService.GetFixAllOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetOperationsCoreAsync(<Unknown Parameters>)
   at Roslyn.Utilities.TaskExtensions.WaitAndGetResult_CanCallOnBackground[T](Task`1 task,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.InvokeWorker(Func`1 getFromDocument,IProgressTracker progressTracker,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.<>c__DisplayClass20_0.<InvokeCore>b__0()
   at Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformAction(IExtensionManager extensionManager,Object extension,Action action)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

_Originally posted by @mavasani in https://github.com/dotnet/roslyn-analyzers/pull/3708#issuecomment-651911519_

**Comment by Youssef1313** [comment]
Should we just check the type before casting?

https://github.com/dotnet/roslyn-analyzers/blob/0743db3bd6950343d127b26b40b303b3fb7b0bfb/src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpPreferIsEmptyOverCount.Fixer.cs#L36

Or there is a deeper issue?

**Comment by Jozkee** [comment]
I don't know your exact scenario but as we discussed, it should be already fixed by #3708, can you please try it out and let me know so we can close this?

**Comment by mavasani** [comment]
Confirmed this is now fixed - https://github.com/dotnet/roslyn-analyzers/pull/3807/files
